### PR TITLE
result of exercise_3

### PR DESCRIPTION
**Pull request by tuongvi91**

### CODE REVIEW

File: exercise/exercise_3.js [hunk 1]
```diff
@@ -1,56 +1,141 @@
...
+console.log(arrays);
+
+// task 2: Add a new user at the end of the array
+arrays.push({ id: 5, name: "user 5", age: 29 })
+console.log(arrays);
+
+// task 3: add multiple users to the array
+addArr1 = [{ id: 6, name: "user 6", age: 18 }];
+addArr2 = [{ id: 7, name: "user 7", age: 28 }];
+newArray = arrays.concat(addArr1, addArr2);
```
Comment (hamongkhang) [inline exercise/exercise_3.js]: Please  add to main array

File: exercise/exercise_3.js [hunk 2]
```diff
@@ -1,56 +1,141 @@
...
+
+// task 5: filter out users with age >= 22;
+result = arrays.filter(function (user) {
+  if (user.age >= 22) { return user; }
+})
+console.log(result);
+
+// task 6: Transform the current array into a new array with the new field "class" of type string
+arrays.map(function (user) {
+  user.class = "itwK";
```
Comment (hamongkhang) [inline exercise/exercise_3.js:50]: cool, I think this mean "IT with Khang"

File: exercise/exercise_3.js [hunk 3]
```diff
@@ -1,56 +1,141 @@
...
+})
+console.log(arrays);
+
+// task 7: Find a user with age = 23
+result = arrays.filter(function (users) {
+  if (users.age === 23) { return users; }
```
Comment (hamongkhang) [inline exercise/exercise_3.js]: you should set "user" not "users"

File: exercise/exercise_3.js [hunk 4]
```diff
@@ -1,56 +1,141 @@
...
+console.log(result);
+
+// task 8: use for loop to get users with age >= 23
+function findUser(array) {
+  var index, count = 0;
+  var newArray = [];
+  for (index = 0; index < arrays.length; index++)
+    if (arrays[index].age >= 23) {
+      newArray[count] = arrays[index];
+      count++;
```
Comment (hamongkhang) [inline exercise/exercise_3.js]: you should try: newArray.push(arrays[index]). Remove count, it's really "phèn"

File: exercise/exercise_3.js [hunk 5]
```diff
@@ -1,56 +1,141 @@
...
+    }
+  return newArray;
+}
+console.log(findUser(arrays));
+
+// task 9: Based on the above information array, and the ids array below, filter out information about users with id as the array below.
+function findUserWithId(id) {
```
Comment (hamongkhang) [inline exercise/exercise_3.js]: findUserById

File: exercise/exercise_3.js [hunk 6]
```diff
@@ -1,56 +1,141 @@
...
+  result = arrays.filter(function (users) {
+    if (users.id === id) { return users; }
```
Comment (hamongkhang) [inline exercise/exercise_3.js]: same: user not users

File: exercise/exercise_3.js [hunk 7]
```diff
@@ -1,56 +1,141 @@
...
+console.log(sortingAsceding(arrays));*/
+result = arrays.sort(function (user1, user2) {
+  return (parseFloat(user1.age) - parseFloat(user2.age))
+})
+console.log(arrays);
 
 // task 11: update the information in the array above with the field "class" converted in task 6 to uppercase (upCase)
+result = arrays.map(function (users) {
+  return users.class.toUpperCase();
+})
```
Comment (hamongkhang) [inline exercise/exercise_3.js]: same with users

File: exercise/exercise_3.js [hunk 8]
```diff
@@ -59,33 +144,92 @@ const person = {
...
-  // expected return result is a string => "Hello User - age: 21"
-};
+//const newFunction = (user) => {
+// handle this function when passing user's params as user's information
+function newFunction(idUser) {
+  var arr = [];
+  arrays.filter(function (users) {
+    if (users.id === idUser) {
+      arr = ("Hello " + users.name + " - age: " + users.age);
+    }
```
Comment (hamongkhang) [inline exercise/exercise_3.js]: same:users->user

File: exercise/exercise_3.js [hunk 9]
```diff
@@ -59,33 +144,92 @@ const person = {
...
+result = arrBoolean.every(function (param) {
+  return (param === true);
+})
+console.log(result);
 
 // task 21: write an object using setter & getter to add new value and get value. key word (Getter, setter)
+const student1 = {
+  gender1: "male",
+  gender2: "female",
+  gender3: "no",
```
Comment (hamongkhang) [inline exercise/exercise_3.js]: haha, male, female and other, not use "no"

File: exercise/exercise_3.js (hunk 2 above)
Comment (tuongvi91) [reply exercise/exercise_3.js:50]: that's right. 555

**Review by hamongkhang — APPROVED** [review]

**Pull request merged**